src/app/dev/DevToys.Tools/Helpers/ListCompareHelper.cs

```diff
@@ -0,0 +1,103 @@
+﻿using DevToys.Tools.Helpers.Core;
+using DevToys.Tools.Models;
+using Markdig.Helpers;
+using Microsoft.Extensions.Logging;
+
+namespace DevToys.Tools.Helpers;
+
+internal static partial class ListCompareHelper
+{
+    public static ResultInfo<string> Compare(
+        string firstList,
+        string secondList,
+        bool caseSensitive,
+        ListComparisonMode comparisonMode,
+        ILogger logger)
+    {
+        try
+        {
+            ResultInfo<string> compareResult;
+            var comparer = new ReadOnlyMemoryEqualityComparer(caseSensitive);
+
+            IEnumerable<ReadOnlyMemory<char>> listA = firstList.AsMemory().ToLines().Where(item => !item.IsEmpty);
```

Comment:
Does it really matter whether the item is not empty? Worse case scenario, the comparer will simply compare that and the HashSet will have an empty item (which is fine), isn't it?

If so, can we remove this `Where` expression? Each call to this method are `O(n)`, which can be significant on a large list.

Reply:
That's right, the problem was on empty files (it always detect as an empty line). But let's try another way.